persist/fs/retriever.go

```diff
@@ -533,6 +532,10 @@ func (req *retrieveRequest) Reset(segment ts.Segment) {
 	req.resultWg.Done()
 }
 
+func (req *retrieveRequest) Clone() xio.Reader {
+	return req
```

Comment:
Hm, it'd be more kosher to:
```go
  req.resultWg.Wait() // wait until result is ready
  if req.err != nil { return nil, err }
  return req.reader.Clone()
```

We probably need to change the `Clone() xio.Reader` interface to be `Clone() (xio.Reader, error)` so callers need to check the error.